src/som/interpreter/actors/Actor.java

```diff
@@ -307,4 +422,91 @@ public String toString() {
       return "Actor[" + (isMain ? "main" : id) + "]";
     }
   }
+
+  public static final class TracingActor extends Actor {
+    protected final long actorId;
+
+    public TracingActor() {
+      super();
+      if (Thread.currentThread() instanceof ActorProcessingThread) {
+        ActorProcessingThread t = (ActorProcessingThread) Thread.currentThread();
+        this.actorId = t.generateActorId();
+      } else {
+        actorId = 0; //main actor
+      }
+    }
+
+    @Override
+    public long getActorId() {
+      return actorId;
+    }
+  }
+
+  public static class Mailbox extends ObjectBuffer<EventualMessage> {
+    public Mailbox(final int bufferSize) {
+      super(bufferSize);
+    }
+
+    public void setExecutionStart(final long start) { }
+    public void setBasemessageId(final long id) { }
+    public void addMessageSendTime() { }
+    public void addMessageExecutionStart() { }
+
+    public long getExecutionStart() { return 0; }
+    public long getBasemessageId() { return 0; }
+    public long getMessageSendTime(final int idx) { return 0; }
+    public long getMessageExecutionStart(final int idx) { return 0; }
+  }
+
+  public static final class TracingMailbox extends Mailbox {
+
+    long baseMessageId;
+    long executionStart;
+    final List<Long> messageExecutionStart = new ArrayList<>();
```

Comment:
If we figure out this is relevant for performance, we should discuss an optimization for it.
But let's see first how it runs with Graal enabled.